### PR TITLE
Support for collapsed menu items and external links

### PR DESCRIPTION
**Pull request by arianics**
screen shot 2018-10-12 at 12 55 19 pm

Pictures are worth a thousand words :)
The "References" section was added to allow for linking to outside resources but also linking back to GitBook or similar documentation projects.

Let us know if we need to make any changes.

Best,

### CODE REVIEW

File: tmpl/layout.tmpl [hunk 1]
```diff
@@ -94,6 +99,15 @@ var usesSearch = Boolean(templates.search);
       <?js= templates.referenceTitle || 'Braintree SDK Client Reference' ?>
     </h3>
 
+    <?js if (resources) { ?>
+      <h3 class="reference-title">
+        References
```
Comment (crookedneighbor) [inline tmpl/layout.tmpl]: Can we call this resources instead of references?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 CHANGELOG
 =========
+3.3.0
```
Comment (crookedneighbor) [inline CHANGELOG.md]: Put `unreleased` here. When we actually do the tagging, we'll change it to `3.3.0`

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jsdoc-template",
-  "version": "3.2.0",
+  "version": "3.3.0",
```
Comment (crookedneighbor) [inline package.json]: Please revert this change.

**Review by crookedneighbor — CHANGES_REQUESTED** [review]
Just a few things:

* Can you go into more detail about what the css fixes are? Looks mostly like you just adjusted some padding and margins, but were there actual problems with the template?
* Can you add documentation for using these new properties (collapse and resources) in the README?

**Comment by arianics** [comment]
Hey @crookedneighbor ,

I just pushed the requested changes.

The "css fixes" refers to the missing `;` in the following snippet. `static/styles/collapse.css` only loads when collapse is enabled and adjusts paddings/margins for this mode.
https://github.com/braintree/jsdoc-template/blob/9ac837f2a774610c3de55431454777ae2419ba11/static/styles/jsdoc-default.css#L362

**Review by crookedneighbor — APPROVED** [review]
Ahh, i see. Should have looked at the file names closer. Somehow totally missed the changes to the default css file.

**Comment by crookedneighbor** [comment]
I think the links should appear on hover, so you don't need to use the parent link to see the child links. If you want to do that as part of this PR, great. If not, we can do it after it's merged.

Let me know which you decide.

**Comment by arianics** [comment]
Let me try it and see how it looks. I’ll let you know today. 

Also, What do you think about keeping the links in mobile menu expanded?

**Comment by crookedneighbor** [comment]
Yeah, it should probably be fully expanded on mobile.

**Comment by arianics** [comment]
I added a media query to collapse.css to only apply the rules on larger displays.

I also have a branch on my forked repo for the hover to expand feature. Let me know what you think.
https://github.com/AgeOfLearning/jsdoc-template/tree/menu-hover

**Comment by crookedneighbor** [comment]
Added in the hover css, but left out the changes to the default css file.

Closed with v3.3.0

**Pull request closed without merging**